public/index.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * GLPI web router.
+ *
+ * This router is used to be able to expose only the `/public` directory on the webserver.
+ */
+
+$glpi_root  = realpath(dirname(__FILE__, 2));
+
+// `$_SERVER['SCRIPT_NAME']` corresponds to the script path relative to server document root.
+// -> if server document root is `/public`, then `$_SERVER['SCRIPT_NAME']` will be equal to `/index.php`
+// -> if script is located into a `/glpi-alias` alias directory, then `$_SERVER['SCRIPT_NAME']` will be equal to `/glpi-alias/index.php`
+$uri_prefix = preg_replace(
+    '/\/index.php$/',
+    '',
+    str_replace('\\', '/', $_SERVER['SCRIPT_NAME'])
+);
```

Comment:
Here we want to strip the `/index.php` from the end of `$_SERVER['SCRIPT_NAME']`, to get only the URI prefix. For instance, if `$_SERVER['SCRIPT_NAME']` is `/my-glpi-alias/index.php`, we want to assign `/my-glpi-alias` in `$uri_prefix`.

Reply:
Yes, `dirname('/my-glpi-alias/index.php')` will give `/my-glpi-alias`

Reply:
infacts, `dirname('/my-glpi-alias/whatever.php') will give `/my-glpi-alias`; that's the only problem I can see (but since there is nothing else expected, it should be good